Mathlib/Algebra/Category/ModuleCat/OverFilteredColimits.lean

```diff
@@ -0,0 +1,351 @@
+/-
+Copyright (c) 2024 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+import Mathlib.Algebra.Ring.Hom.Defs
+import Mathlib.Algebra.Module.Defs
+import Mathlib.CategoryTheory.Limits.ConcreteCategory
+import Mathlib.CategoryTheory.ConcreteCategory.Elementwise
+
+/-!
+# Module structures of filtered colimits of abelian groups over filtered colimts of rings
+
+Let `R` be the filtered colimit of rings `{Rⱼ}` and `M` be the filtered colimit of
```

Comment:
My intuition is no. For example, the coproduct in the category of noncommutative ring behaves quite badly but coproduct in the category of abelian group is fine. The result in this file says the module structure shouldn't depend on whether the colimit is in RingCat or CommRingCat, but since colimit if not filtered generally will be different in RingCat and CommRingCat, it might still be possible to setup a module structure, but we will need to say which category we are talking about.

Reply:
But are not you considering only filtered colimits?